packages/react/src/profiler.tsx

```diff
@@ -0,0 +1,124 @@
+ * even if RAF doesn't fire (for example if the browser tab is not visible)
+ *
+ * This is what we use to tell if a component activity has finished
+ *
+ */
+function afterNextFrame(callback: Function): void {
+  let timeout: number | undefined;
+  let raf: number;
+
+  const done = () => {
+    window.clearTimeout(timeout);
+    window.cancelAnimationFrame(raf);
+    window.setTimeout(callback);
+  };
+
+  raf = window.requestAnimationFrame(done);
+  timeout = window.setTimeout(done, 100);
+}
+
+const getInitActivity = (componentDisplayName: string): number | null => {
+  const tracingIntegration = getCurrentHub().getIntegration(TRACING_GETTER);
+
+  if (tracingIntegration !== null) {
+    // tslint:disable-next-line:no-unsafe-any
+    const activity = (tracingIntegration as any).constructor.pushActivity(componentDisplayName, {
+      data: {},
+      description: `<${componentDisplayName}>`,
+      op: 'react',
+    });
+
+    // tslint:disable-next-line: no-unsafe-any
+    return activity;
+  }
+
+  logger.warn(
+    `Unable to profile component ${componentDisplayName} due to invalid Tracing Integration. Please make sure to setup the Tracing integration.`,
+  );
+  return null;
+};
+
+interface ProfilerProps {
+  componentDisplayName?: string;
+}
+
+class Profiler extends React.Component<ProfilerProps> {
+  public activity: number | null;
+
+  public constructor(props: ProfilerProps) {
+    super(props);
+
+    const { componentDisplayName = UNKNOWN_COMPONENT } = this.props;
+
+    this.activity = getInitActivity(componentDisplayName);
+  }
+
+  public componentDidMount(): void {
+    afterNextFrame(this.finishProfile);
+  }
+
+  public componentWillUnmount(): void {
+    afterNextFrame(this.finishProfile);
+  }
+
+  public finishProfile = () => {
+    if (!this.activity) {
+      return;
+    }
+
+    const tracingIntegration = getCurrentHub().getIntegration(TRACING_GETTER);
+    if (tracingIntegration !== null) {
+      // tslint:disable-next-line:no-unsafe-any
+      (tracingIntegration as any).constructor.popActivity(this.activity);
+      this.activity = null;
+    }
+  };
+
+  public render(): React.ReactNode {
+    return this.props.children;
+  }
+}
+
+function withProfiler<P extends object>(
+  WrappedComponent: React.ComponentType<P>,
+  profilerProps?: ProfilerProps,
+): React.FC<P> {
+  const componentDisplayName = WrappedComponent.displayName || WrappedComponent.name || UNKNOWN_COMPONENT;
+
+  const Wrapped: React.FC<P> = (props: P) => (
+    <Profiler componentDisplayName={componentDisplayName} {...profilerProps}>
+      <WrappedComponent {...props} />
+    </Profiler>
+  );
+
+  Wrapped.displayName = `profiler(${componentDisplayName})`;
+
+  // Copy over static methods from Wrapped component to Profiler HOC
+  // See: https://reactjs.org/docs/higher-order-components.html#static-methods-must-be-copied-over
+  hoistNonReactStatic(Wrapped, WrappedComponent);
```

Comment:
I've decided to keep this library for now instead of vendoring it in. This is as it leverages functionality from `react-is`, https://github.com/facebook/react/tree/master/packages/react-is, which is hard to keep up to date properly (lot's of manual copy-paste, hard to know when different react types will change).

Many popular libraries use `hoist-non-react-statics`. See:
`react-intl`: https://github.com/formatjs/formatjs/blob/master/packages/react-intl/package.json#L143
`react-redux`: https://github.com/reduxjs/react-redux/blob/master/package.json#L52
`react-apollo`: https://github.com/apollographql/react-apollo/blob/master/packages/hoc/package.json#L45

Reply:
I think it's worth the bundle size change here for a much smoother experience, it adds `1.2kb` gzipped + minified - https://bundlephobia.com/result?p=hoist-non-react-statics@3.3.2

Reply:
OK, let's keep it for now.